src/Compilers/Core/Portable/InternalUtilities/SpecializedCollections.Symbol.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Roslyn.Utilities
+{
+    internal static partial class SpecializedCollections
+    {
+        public static PooledHashSet<ISymbol> GetPooledSymbolHashSetInstance<TSymbol>() where TSymbol : ISymbol
```

Comment:
>GetPooledSymbolHashSetInstance [](start = 45, length = 30)

Should this method and `s_poolInstance` be moved to `PooledHashSet<T>`?

Reply:
I guess this needs to be a generic method in a non-generic type. Perhaps move to a non-generic `PooledHashSet` class.

---
In reply to: [305443467](https://github.com/dotnet/roslyn/pull/37247#discussion_r305443467) [](ancestors = 305443467)

Reply:
You can't, because PooledHashSet doesn't reference ISymbol (it's at the 'top' of the hierarchy) so we have to have it at the same level as ISymbol